records/generator.go

```diff
@@ -232,24 +258,41 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 				continue
 			}
 
-			tname := labels.AsDNS952(task.Name)
-			sid := slaveIdTail(task.SlaveId)
-			tag := hashString(task.Id)
-			tail := fname + "." + domain + "."
+			// context used to resolve DomainPattern variables for the current task
+			context := PatternContext{
+				SlaveId: slaveIdTail(task.SlaveId),
+				TaskId: hashString(task.Id),
```

Comment:
can we incorporate additional information into this hash? (and maybe modify the hash to actually use (at least) 32 bits). for example hashString(task.Id+task.SlaveId+task.frameworkId+task.executorId). and if we do that, do we really need the tail(slaveId) anymore?